x-pack/plugins/profiling/public/components/subchart.tsx

```diff
@@ -16,35 +14,98 @@ import {
 import { CountPerTime } from '../../common/topn';
+import { useProfilingParams } from '../hooks/use_profiling_params';
+import { useProfilingRouter } from '../hooks/use_profiling_router';
+import { chartTheme } from '../utils/chart_styles';
+import { asPercentage } from '../utils/formatters/as_percentage';
 
 export interface SubChartProps {
-  id: string;
-  name: string;
+  index: number;
+  color: string;
   height: number;
+  width?: number;
+  category: string;
+  percentage: number;
   data: CountPerTime[];
-  x: string;
-  y: string;
 }
 
-export const SubChart: React.FC<SubChartProps> = ({ id, name, height, data, x, y }) => {
+export const SubChart: React.FC<SubChartProps> = ({
+  index,
+  color,
+  category,
+  percentage,
+  height,
+  data,
+  width,
+}) => {
+  const theme = useEuiTheme();
+
+  const profilingRouter = useProfilingRouter();
+
+  const { path, query } = useProfilingParams('/stacktraces/{topNType}');
+
+  const href = profilingRouter.link('/stacktraces/{topNType}', {
+    path: {
+      topNType: TopNType.Traces,
+    },
+    query: {
+      ...query,
+      kuery: `${getFieldNameForTopNType(path.topNType)}:"${category}"`,
+    },
+  });
+
   return (
-    <Chart size={{ height }}>
-      <Settings showLegend={false} />
-      <AreaSeries
-        id={id}
-        name={name}
-        data={data}
-        xAccessor={x}
-        yAccessors={[y]}
-        xScaleType={ScaleType.Time}
-        yScaleType={ScaleType.Linear}
-        areaSeriesStyle={{ area: { opacity: 0.3 }, line: { opacity: 1 } }}
-        curve={CurveType.CURVE_STEP_AFTER}
-      />
-      <Axis id="bottom-axis" position="bottom" tickFormat={timeFormatter('YYYY-MM-DD HH:mm:ss')} />
-      <Axis id="left-axis" position="left" showGridLines tickFormat={(d) => Number(d).toFixed(0)} />
-    </Chart>
+    <EuiFlexGroup direction="column">
+      <EuiFlexItem>
+        <EuiFlexGroup direction="row" gutterSize="m" alignItems="flexStart">
+          <EuiFlexItem grow={false}>
+            <EuiBadge color={color}>
+              <EuiText color={theme.euiTheme.colors.lightestShade} size="xs">
+                {index}
+              </EuiText>
+            </EuiBadge>
+          </EuiFlexItem>
+          <EuiFlexItem grow>
+            <EuiLink href={href}>
+              <EuiText size="s">{category}</EuiText>
+            </EuiLink>
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiText size="s">{asPercentage(percentage / 100, 2)}</EuiText>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      </EuiFlexItem>
+      <EuiFlexItem>
+        <Chart size={{ height, width }}>
+          <Settings showLegend={false} tooltip={{ showNullValues: false }} theme={chartTheme} />
+          <AreaSeries
+            id={category}
+            name={category}
+            data={data}
+            xAccessor={'Timestamp'}
+            yAccessors={['Count']}
+            xScaleType={ScaleType.Time}
```

Comment:
When you use time charts is always recommended to specify the `timeZone` prop to nicely match that to the timezone of the data/local etc.